Add contributing file (#307)
bw2/ConfigArgParse: CONTRIBUTING.rst

Contributing to ConfigArgParse
------------------------------

What can you do 
~~~~~~~~~~~~~~~

If you like the project and think you could help with making it better, there are many ways you can do it:

> - Create a new issue for new feature proposal or a bug

Reviewer: ```suggestion
- Create an issue to report a bug or suggest a new feature
```

> - Triage old issues that needs a refresh

Reviewer: ```suggestion
- Triage old issues that need a refresh
```

- Implement existing issues
- Help with improving the documentation

> - Spread a word about the project to your colleagues, friends, blogs or any other channels

Reviewer: ```suggestion
- Spread the word about the project to your colleagues, friends, blogs or any other channels
```

- Any other things you could imagine

Any contribution would be of great help and we'll highly appreciate it! 
If you have any questions, please create a new issue.

Development process
~~~~~~~~~~~~~~~~~~~

Create a fork of the git repository and checkout a new branch from master branch. 
The branch name may start with an associated issue number so that we can easily cross-reference them. 
For example, use ``1234-some-brach-name`` as the name of the branch working to fix issue 1234. 

> Once you’re ready to run a full batterie of tests to your changes, open a pull request.

Reviewer: ```suggestion
Once your changes are ready and are passing the automated tests, open a pull request.
```

Don’t forget to sync your fork once in while to work from the latest revision.

Pre-commit checks
~~~~~~~~~~~~~~~~~

> - Run unit tests: ``pytest``. 

Reviewer 2: I think you miss my point. The existing tests in this project do *not* actually use pytest, they use *unittest*. It's just that the `pytest` command has a feature of also picking up and running the *unittest* tests. I'm not denying that pytest has good features, just noting that it's not necessary to install the pytest package to run the existing test suite.

> - Run unit tests: ``pytest``. 

Reviewer 3: Its best to stay with a single testing framework that everyone uses, and not mix and matching. Just because `python -m unittest` works now, doesn't mean that it will work tomorrow.

> - Run unit tests: ``pytest``. 

Reviewer 3: I think my point I was trying to make is that the project is using `pytest` today, and thus it will work tomorrow. I'm planning to add (if the powers that be allow me) the `xdist` plugin to make the tests run faster -- that will most definitely break unittest.

> - Run unit tests: ``pytest``. 

Reviewer 2: Your statement is factually incorrect and a non-sequitur. I don't even have the pytest module installed, and all the tests still run. The CI environment does run the pytest command but you can switch it to run `python -munittest` directly, drop the pytest dependency, and nothing changes. It's the unittest module that is running the tests under the hood, and it always has been.

Similarly I can install nose2 and run all the tests by running `nose2`. It works, but this doesn't mean that the tests in this module are "using" nose2.

The entire test suite runs on my 10 year old laptop in 3.39 seconds, so I'd personally be sceptical that it needs new dependencies on special plugins to speed it up.

> - Run unit tests: ``pytest``. 

Reviewer 3: I see this is a contentious point for you. I am not the maintainer, and I don't have enough skin in the game for it to matter either way. I'll leave this to the maintainer. Best of luck.

Review process 
~~~~~~~~~~~~~~

- Code changes and code added should have tests: untested code is buggy code and should

>   not be accepted by reviewers.

Reviewer 4: Good point. I think creating a simple example of adding a unit test based on a past issue might make sense. Maybe the `--` arg separator is a good example?

>   not be accepted by reviewers.

Reviewer 2: Cool. Rather than hold up this PR just now I'll just make an issue to add this later.

> - All code changes must be reviewed by at least one maintainers who is not an author 

Reviewer: ```suggestion
- All code changes must be reviewed by at least one maintainer who is not an author 
```

  of the code being added.

> - When one is done with the review, always say what the next step(s) should be: 

Reviewer: ```suggestion
- When a reviewer completes a review, they should always say what the next step(s) should be: 
```
